cli/main.rs

```diff
@@ -1,29 +1,10 @@
 use std::env;
 
 use clap::Parser;
-use sp_runtime::{generic, traits::BlakeTwo256};
-use subxt::PolkadotConfig;
+use node_executor::ExecutorDispatch;
+use node_primitives::Block;
 use try_runtime_core::commands::TryRuntime;
 
-#[subxt::subxt(
-    runtime_metadata_url = "wss://rpc.polkadot.io:443",
-    derive_for_all_types = "Clone, PartialEq, Eq",
-    runtime_types_only
-)]
-pub mod polkadot {}
-
-type RuntimeCall = polkadot::runtime_types::polkadot_runtime::RuntimeCall;
-
-type UncheckedExtrinsic = generic::UncheckedExtrinsic<
-    <PolkadotConfig as subxt::Config>::Address,
-    RuntimeCall,
-    <PolkadotConfig as subxt::Config>::Signature,
-    (),
->;
-
-type Header = generic::Header<u32, BlakeTwo256>;
-type Block = generic::Block<Header, UncheckedExtrinsic>;
```

Comment:
This code was not compiling with latest substrate and latest subxt, issues with some `subxt` trait bounds.

I'm not sure exactly why it was necessary. @kianenigma suggested in https://github.com/paritytech/substrate/issues/12975 that it's needed to fetch the block format for the chain being used but I have had no issues just using `node_primitives::Block;` across polkadot and testnets.

Perhaps it is necessary for `fast-forward`? If that's the case, `fast-forward` shouldn't block us from moving to a standalone CLI (it doesn't work already on Substrate and also we need to resolve the inherents issue separately), so I propose we first migrate stable features to this standalone cli, deprecate it in substrate, and afterwards add `fast-forward` support here.

Reply:
@kianenigma was right - in order to be chain-agnostic, `try-runtime-cli` has to be generic over the block format; while it was a subcommand of the node binary, the block type was already given in the context, and thus there was no need for `subxt` support

once we go out from node binary, we must have a way of calling the target chain and getting the proper block type; `node_primitives::Block` might be working for most chains, since it is kinda default one, but it will fail for any chain that is configured differently...

IMHO we can remove it for the sake of this PR, but this ought to be revisited at some point

Reply:
https://github.com/paritytech/try-runtime-cli/issues/18